pkg/awsiamauth/hook.go

```diff
@@ -0,0 +1,51 @@
+package awsiamauth
```

Comment:
What do you think about moving this to `pkg/awsiamauth/hook/`?
It avoids importing the workload packages from the main business logic packages.

It's not necessary with this one in particular but I feel it sets a good precedence for the future.

Reply:
I don't think its necessary in this case. Equally, I could see it being appropriate in other cases.

In general, I think its good to stick to the rule "package A provides everything to do with A" and only break it down if needed.

Reply:
We agreed on my response above.